src/main/java/com/terraforged/mod/server/command/TerraCommand.java

```diff
@@ -309,10 +351,40 @@ private static CommandSyntaxException createException(String type, String messag
 
     private static ITextComponent createTeleportMessage(BlockPos pos) {
         return TextComponentUtils.wrapInSquareBrackets(new TranslationTextComponent(
-                "chat.coordinates", pos.getX(), "~", pos.getZ()
+                "chat.coordinates", pos.getX(), pos.getY(), pos.getZ()
         )).applyTextStyle((style) -> style.setColor(TextFormatting.GREEN)
                 .setClickEvent(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, "/tp @s " + pos.getX() + " " + pos.getY() + " " + pos.getZ()))
                 .setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new TranslationTextComponent("chat.coordinates.tooltip")))
         );
     }
+
+    private static ITextComponent createIdentifierMessage(int identifier) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(TextComponentUtils.wrapInSquareBrackets(new StringTextComponent(
+                        String.format("%03d", identifier)
+                   )).applyTextStyle(SEARCH_ID_FORMAT));
+    }
+
+    private static ITextComponent createTerrainMessage(Terrain terrain) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(new StringTextComponent(terrain.getName()).applyTextStyle(DISPLAY_NAME_FORMAT));
+    }
+
+    private static ITextComponent createBiomeDisplayMessage(Biome biome) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(biome.getDisplayName())
+                       .applyTextStyle((style) -> style.setColor(DISPLAY_NAME_FORMAT)
+                                                       .setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, createBiomeRegistryMessage(biome)))
+                       );
+    }
+
+    private static ITextComponent createBiomeRegistryMessage(Biome biome) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(new StringTextComponent("" + biome.getRegistryName()).applyTextStyle(INTERNAL_NAME_FORMAT));
+    }
+
+    private static ITextComponent createBiomeTypeMessage(BiomeType biomeType) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(new StringTextComponent(biomeType.name()).applyTextStyle(INTERNAL_NAME_FORMAT));
+    }
```

Comment:
Are you wanting to consolidate all the names into a single method/format? If it's going to be a single method/format, I'd be inclined to make it the `ITALICS` rather than colored, and remove the `Biome` display name option.

Reply:
Yeah so instead of having a creatMessage for biome, biome registry name, terrain, biome type there can just be one method that formats the string with the same styling - they're the subject of the search/query so should have equal formatting.

I don't mind what the formatting is if any tbh . These commands are there for my use primarily - I'm not too concerned how they look 😅